internal/devconfig/init.go

```diff
@@ -33,6 +35,30 @@ func Init(dir string, writer io.Writer) (created bool, err error) {
 			color.HiYellowString(s),
 		)
 	}
+	return created, err
+}
+
+func initConfigFile(path string) (created bool, err error) {
+	file, err := os.OpenFile(path, os.O_RDWR|os.O_CREATE|os.O_EXCL, 0o644)
+	if errors.Is(err, os.ErrExist) {
+		return false, nil
+	}
+	if err != nil {
+		return false, err
+	}
+	defer func() {
+		if err != nil {
+			os.Remove(file.Name())
```

Comment:
This is only for initializing a new `devbox.json`. The file will only be created if it doesn't already exist (because of `os.O_EXCL`). If something goes wrong I think it makes sense to delete it instead of leaving a partially written `devbox.json`.